src/charm.py

```diff
@@ -59,63 +58,54 @@ def _setup_database_requirer(self, relation_name: str, database_name: str) -> Da
         return database_requirer
 
-    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
-        """Event triggered when a database was created for this application.
-
-        Args:
-            event: The DatabaseCreatedEvent object
-        """
-        logger.debug("Database credentials are received: %s", event.username)
-        self.reconciliation(event)
-
-    def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
-        """Handle relation broken event.
-
-        Args:
-            event: The RelationBrokenEvent object
-        """
-        logger.debug("Relation with database removed: %s", event.relation.name)
-        self.reconciliation(event)
-
     def _datasource(self) -> dict[str, str]:
         """Compute datasource dict and return it.
 
         Returns:
             Dict containing details about the data provider integration
         """
-        if self.database:
-            relations_data = list(self.database.fetch_relation_data().values())
-            if relations_data:
-                # There can be only one database integrated at a time
-                # cf: metadata.yaml
-                data = relations_data[0]
-                if "endpoints" in data and "username" in data and "password" in data:
-                    # We assume that the relation follows the following json schema:
-                    # https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/mysql_client/v0/schemas/provider.json
-                    database_name = data.get("database", self.database.database)
-                    endpoint = data["endpoints"].split(",")[0]
-                    return {
-                        "username": data["username"],
-                        "password": data["password"],
-                        "url": f"jdbc:mysql://{endpoint}/{database_name}",
-                    }
-        return {
+        # Default results when something went wrong
+        default = {
             "username": "",
             "password": "",
             "url": "",
         }
 
+        relations_data = list(self.database.fetch_relation_data().values())
+
+        if not relations_data:
+            logger.warning("No relation data from data provider: %s", self.database.database)
+            return default
+
+        # There can be only one database integrated at a time
+        # cf: metadata.yaml
+        data = relations_data[0]
+
+        # Let's check that the relation data is well formed according to the following json_schema:
+        # https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/mysql_client/v0/schemas/provider.json
+        if not all(data.get(key) for key in ("endpoints", "username", "password")):
+            logger.warning("Incorrect relation data from the data provider: %s", data)
+            return default
+
+        database_name = data.get("database", self.database.database)
```

Comment:
get `database` from `database.database` - kind of confusing?

Reply:
We could rename the class attribute to `database_provider` or `db_provider`